mbed_devices/_internal/resolve_target.py

```diff
@@ -50,7 +43,10 @@ def _build_target_resolver(candidate: CandidateDevice) -> Callable:
         online_id = _extract_online_id(all_files_contents)
         return functools.partial(get_target_by_online_id, slug=online_id.device_slug, target_type=online_id.device_type)
     except OnlineIdNotFound:
-        raise UnableToBuildResolver
+        pass
+
+    # Product code might be the first 4 characters of the serial number
```

Comment:
```suggestion
    # Check if the first 4 characters of the serial number is the product code. It is only in some cases (eg. ST boards).
```
(I am not sure if ST boards are, just for example).

Reply:
Hmm, actually maybe this is worse. Its' not really a check. I retract my comment!